pkg/cable/metrics.go

```diff
@@ -55,32 +60,126 @@ var (
-		remoteClusterLabel:  remoteEndpoint.ClusterID,
-		remoteHostnameLabel: remoteEndpoint.Hostname,
+		cableDriverLabel:      cableDriverName,
+		localClusterLabel:     localEndpoint.ClusterID,
+		localHostnameLabel:    localEndpoint.Hostname,
+		localEndpointIpLabel:  localEndpoint.PublicIP,
+		remoteClusterLabel:    remoteEndpoint.ClusterID,
+		remoteHostnameLabel:   remoteEndpoint.Hostname,
+		remoteEndpointIpLabel: remoteEndpoint.PublicIP,
 	}).Set(float64(bytes))
 }
 
 func RecordTxBytes(cableDriverName string, localEndpoint, remoteEndpoint *submv1.EndpointSpec, bytes int) {
 	txGauge.With(prometheus.Labels{
-		cableDriverLabel:    cableDriverName,
-		localClusterLabel:   localEndpoint.ClusterID,
-		localHostnameLabel:  localEndpoint.Hostname,
-		remoteClusterLabel:  remoteEndpoint.ClusterID,
-		remoteHostnameLabel: remoteEndpoint.Hostname,
+		cableDriverLabel:      cableDriverName,
+		localClusterLabel:     localEndpoint.ClusterID,
+		localHostnameLabel:    localEndpoint.Hostname,
+		localEndpointIpLabel:  localEndpoint.PublicIP,
+		remoteClusterLabel:    remoteEndpoint.ClusterID,
+		remoteHostnameLabel:   remoteEndpoint.Hostname,
+		remoteEndpointIpLabel: remoteEndpoint.PublicIP,
 	}).Set(float64(bytes))
 }
+
+func RecordConnectionLatency(cableDriverName string, localEndpoint, remoteEndpoint *submv1.EndpointSpec, latencySeconds float64) {
+	connectionLatencySecondsGague.With(prometheus.Labels{
+		cableDriverLabel:      cableDriverName,
+		localClusterLabel:     localEndpoint.ClusterID,
+		localHostnameLabel:    localEndpoint.Hostname,
+		localEndpointIpLabel:  localEndpoint.PublicIP,
+		remoteClusterLabel:    remoteEndpoint.ClusterID,
+		remoteHostnameLabel:   remoteEndpoint.Hostname,
+		remoteEndpointIpLabel: remoteEndpoint.PublicIP,
+	}).Set(latencySeconds)
+}
+
+func RecordConnectionStatusActive(cableDriverName string, localEndpoint, remoteEndpoint *submv1.EndpointSpec) {
+	labels := prometheus.Labels{
+		cableDriverLabel:      cableDriverName,
+		localClusterLabel:     localEndpoint.ClusterID,
+		localHostnameLabel:    localEndpoint.Hostname,
+		localEndpointIpLabel:  localEndpoint.PublicIP,
+		remoteClusterLabel:    remoteEndpoint.ClusterID,
+		remoteHostnameLabel:   remoteEndpoint.Hostname,
+		remoteEndpointIpLabel: remoteEndpoint.PublicIP,
+	}
+	connectionStatusGauge.With(labels).Set(float64(1))
+	connectionEstablishedTimestampGauge.With(labels).Set(float64(time.Now().Unix()))
+}
+
+func RecordConnectionStatusInactive(cableDriverName string, localEndpoint, remoteEndpoint *submv1.EndpointSpec) {
+	labels := prometheus.Labels{
+		cableDriverLabel:      cableDriverName,
+		localClusterLabel:     localEndpoint.ClusterID,
+		localHostnameLabel:    localEndpoint.Hostname,
+		localEndpointIpLabel:  localEndpoint.PublicIP,
+		remoteClusterLabel:    remoteEndpoint.ClusterID,
+		remoteHostnameLabel:   remoteEndpoint.Hostname,
+		remoteEndpointIpLabel: remoteEndpoint.PublicIP,
+	}
+	txGauge.Delete(labels)
+	rxGauge.Delete(labels)
```

Comment:
I can’t find the details in the docs right now, but I think that deleting results in the gauge not being exported for those labels, and that might not be what we want for rx/tx. I’ll check tomorrow and update.

Reply:
shouldn't we use same labels for rx/tx gauges as all gauges in this file?